src/Sentry/Internal/SettingLocator.cs

```diff
@@ -33,16 +33,34 @@ public SettingLocator(SentryOptions options)
     public string GetDsn()
     {
         // For DSN only
-        // An empty string set on the option should NOT be converted to null because it is used
-        // to indicate the the SDK is disabled.
-        _options.Dsn ??= GetEnvironmentVariable(Constants.DsnEnvironmentVariable)
```

Comment:
I'd just do

```cs
if (String.IsNullOrEmpty(_options.Dsn))
{
    _options.Dsn = GetEnvironmentVariable(Constants.DsnEnvironmentVariable)
                         ?? AssemblyForAttributes?.GetCustomAttribute<DsnAttribute>()?.Dsn;
}
```

Reply:
Yeah, I thought about that.
But if we want to keep the "fix" for the "Hey, I'm a new user, I put `.UseSentry` somewhere but nothing happens. What do?` experience, then we need to check for those separately. Otherwise, it'll still throw if you explicitly set `options.Dsn = string.Empty` and there is nothing in the environment.

Reply:
I'm not sure I follow you. null will remain null if neither env nor assembly attribute was set.